### PR TITLE
Add request/response contract to API.md

### PR DESCRIPTION
**Pull request by dhinklexo**
Add request/response contract to API.md

## Types of changes

- [x] Documentation only (no changes to either `lib/` or `test/` files)
- [ ] Bug fix (non-breaking change which fixes an issue. you didn't modify existing tests)
- [ ] New feature (non-breaking change which adds functionality. you added at least one new test)
- [ ] Breaking change (fix or feature that would cause existing functionality to change. you had to modify existing tests.)

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/xogroup/toki/blob/master/.github/CONTRIBUTING.md) document.
- [ ] My code follows the [Hapi.js style guide](https://github.com/hapijs/contrib/blob/master/Style.md).
- [x] I have updated the documentation as needed.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -102,18 +105,43 @@ toki.on('error', (error)=>{
...
+Request is a decorated version of the Node http-server request object. It will always have the following:
+
++ `request.query` - a parsed query object
++ `request.params` - an object of any params from passed paths
++ `request.path` - the current path
++ `request.method` - the method which called this request
++ `request.headers` - an object containing all headers
+
+
+#### Response
```
Comment (WesTyler) [inline API.md]: @dhinklexo does calling something like `response.send(payload)` from within the mdthod (like toki-method-http) simply attach that payload to the response that will eventually be sent by Toki? Or does it actually fire off the response object back to the client?

Basically do the toki plugins act like hapi prerequest handlers/decorators, or like the actual request handlers?

Comment (WesTyler) [inline API.md]: Or is this just a response object that is sent back to Toki to map to the main response and runtime context?

Comment (dhinklexo) [inline API.md]: (Context)
This is just a copy of the bridge response/request API

(Answer)
It looks like that IS the current behavior. Whether or not that's the desired behavior on the other hand..

Comment (WesTyler) [inline API.md]: My understanding from our initial talks about `toki-method-http` and `toki-method-rabbit` is that the method should define the shape of its result (via templated config) and do 2 things:
  1. pass that response back to toki, which will add it to the runtime context under the action's `name` property. future potential actions now have access to this mapped result.
  2. modify the "main" response object without sending it back to the client. This way toki can build up the main response by running it through all of the methods, then at the end send the total hydrated response back to the requesting client.

Comment (WesTyler) [inline API.md]: Moved the discussion [here](https://github.com/xogroup/toki-discussion/issues/1)

**Pull request merged**